theano/sandbox/cuda/dnn.py

```diff
@@ -1710,6 +1707,24 @@ def c_code(self, node, name, inputs, outputs, sub):
   %(fail)s
 }
 
+// if input batch is empty, we return the empty output without calling cuDNN
+// (which will fail on zero batch size)
+if (CudaNdarray_DIMS(%(input)s)[0] == 0) {
+  cudaError_t err2 = cudaMemset((%(out)s)->devdata, 0,
+                                CudaNdarray_SIZE(%(out)s) * sizeof(real));
```

Comment:
This call is useless, as out will have a shape of (0, ....) and so the total array size is 0.